src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,220 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A general domain for two compact objects.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain that represents a binary compact object solution. The
+ * Domain consists of four/five nested layers of blocks; these layers are,
+ * working from the interior toward the exterior:
+ * - 0: (optionally) The block at the center of each compact object, if not
+ *      excised. If present, this block is a cube. If excised, the hole left
+ *      by its absence is spherical.
+ * - 1: The blocks that resolve each individual compact object. This layer has
+ *      a spherical outer boundary - if the corresponding layer-0 block exists,
+ *      then the layer is a cube-to-sphere transition; if the layer-0 block is
+ *      excised, then the layer is a spherical shell.
+ * - 2: The blocks that surround each object with a cube. Around each compact
+ *      object, this layer transitions from a sphere to a cube.
+ * - 3: The blocks that surround each cube with a half-cube. At this layer, the
+ *      two compact objects are enclosed in a single cube-shaped grid.
+ * - 4: The blocks that form the outer sphere. This layer transitions back to
+ *      spherical and can extend to large radial distances from the compact
+ *      objects.
+ * In the code and options below, `ObjectA` and `ObjectB` refer to the two
+ * compact objects, and by extension, also refer to the layers that immediately
+ * surround each compact object. Note that `ObjectA` is located to the left of
+ * the origin (along the negative x-axis) and `ObjectB` is located to the right
+ * of the origin.
+ * `enveloping cube` and `enveloping sphere` refer to the outer surfaces of
+ * layers 4 and 5 respectively. Both of these surfaces are centered at the
```

Comment:
I see only layers 0 thru 4 above.  What is layer 5?

Reply:
Oh that's my bad :(

Reply:
I should have caught this, thank you!